src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs

```diff
@@ -1013,7 +1013,7 @@ string wasUpdatedCheck(Field field)
                                 : TypeIsSymbol(field)
                                     ? "!Symbols.SymbolEqualityComparer.ConsiderEverything.Equals({0}, this.{1})"
                                     : IsValueType(field.Type) && field.Type[^1] == '?'
-                                        ? "{0}.Equals(this.{1})"
+                                        ? "!{0}.Equals(this.{1})"
```

Comment:
`Nullable<T>.Equals(object)` requires boxing the argument. Perhaps we should use an alternative.